liquibase-core/src/main/java/liquibase/snapshot/jvm/UniqueConstraintSnapshotGenerator.java

```diff
@@ -176,24 +177,34 @@ protected void addTo(DatabaseObject foundObject, DatabaseSnapshot snapshot) thro
                 }
                 sql += "order by ordinal_position";
             } else if (database instanceof PostgresDatabase) {
+                List<String> conditions = new ArrayList<>();
                 sql = "select const.CONSTRAINT_NAME, COLUMN_NAME, const.constraint_schema as CONSTRAINT_CONTAINER "
                         + "from " + database.getSystemSchema() + ".table_constraints const "
                         + "join " + database.getSystemSchema() + ".key_column_usage col "
                         + "on const.constraint_schema=col.constraint_schema "
                         + "and const.table_name=col.table_name "
-                        + "and const.constraint_name=col.constraint_name "
-                        + "where const.constraint_catalog='" + database.correctObjectName(schema.getCatalogName(), Catalog.class) + "' ";
+                        + "and const.constraint_name=col.constraint_name ";
+                if (schema.getCatalogName() != null) {
```

Comment:
Two problems solved here: 
* When catalog name and schema name were missing, the query would try to look for a catalog/schema with the literal name `'null'`
* Missing space between `const.constraint_name=XXX` and `order by`